django_project/geohosting/serializer/product.py

```diff
@@ -34,9 +34,19 @@ class ProductDetailSerializer(serializers.ModelSerializer):
     packages = serializers.SerializerMethodField()
     product_meta = serializers.SerializerMethodField()
 
+    def __init__(self, *args, **kwargs):
+        self.currency = kwargs.pop('currency', None)
+        super().__init__(*args, **kwargs)
+
     def get_packages(self, obj: Product):
         packages = obj.packages.all()
```

Comment:
Hi @tinashechiraya 
You can just filter it here, so we make sure it is the correct currency
By default is ZAR if no currency shown